app/server.go

```diff
@@ -727,11 +728,28 @@ func runSecurityJob(s *Server) {
 	}, time.Hour*4)
 }
 
+func doDiagnosticsIfNeeded(s *Server, firstRun time.Time) {
+	hoursSinceFirstServerRun := time.Since(firstRun).Hours()
+	// Send once every 10 minutes for the first hour
+	// Send once every hour thereafter for the first 12 hours
+	// Send at the 24 hour mark and every 24 hours after
+	if hoursSinceFirstServerRun < 1 {
+		doDiagnostics(s)
+	} else if hoursSinceFirstServerRun <= 12 && time.Since(s.timestampLastDiagnosticSent) >= time.Hour {
+		doDiagnostics(s)
+	} else if hoursSinceFirstServerRun > 12 && time.Since(s.timestampLastDiagnosticSent) >= 24*time.Hour {
+		doDiagnostics(s)
+	}
+}
+
 func runDiagnosticsJob(s *Server) {
+	// Send on boot
 	doDiagnostics(s)
+	// we ignore the error, since it's been ensured on server start
+	firstRun, _ := s.FakeApp().getFirstServerRunTimestamp()
```

Comment:
Can still fail though to query the database though. What would the intended behaviour be in that case? I'd prefer to see the error handled and a sensible default value for `firstRun` be set based on that - it feels more explicit that way.

Reply:
done.